File: Runtime/TransitionOrder.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'************************************************************************************

Friend Class TransitionOrder

    Private m_TransitionGroupId As Integer
    Private m_Iteration As Nullable(Of Integer)
    Private m_Timestep As Nullable(Of Integer)
    Private m_Order As Double = DEFAULT_TRANSITION_ORDER

    Public Sub New(
        ByVal transitionGroupId As Integer,
        ByVal iteration As Nullable(Of Integer),
        ByVal timestep As Nullable(Of Integer),
        ByVal order As Nullable(Of Double))

        Me.m_TransitionGroupId = transitionGroupId
        Me.m_Iteration = iteration
        Me.m_Timestep = timestep

        If (order.HasValue) Then
            Me.m_Order = order.Value
        End If

    End Sub

    Public ReadOnly Property TransitionGroupId As Integer
        Get
            Return Me.m_TransitionGroupId
        End Get
    End Property

    Public ReadOnly Property Iteration As Nullable(Of Integer)
        Get
            Return Me.m_Iteration
        End Get
    End Property

    Public ReadOnly Property Timestep As Nullable(Of Integer)
        Get
            Return Me.m_Timestep
        End Get
    End Property

    Public ReadOnly Property Order As Double
        Get
            Return Me.m_Order
        End Get
    End Property

End Class

File: Runtime/TransitionOrder.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

namespace SyncroSim.STSim
{
    internal class TransitionOrder
    {
        private int m_TransitionGroupId;
        private int? m_Iteration;
        private int? m_Timestep;
        private double m_Order = Constants.DEFAULT_TRANSITION_ORDER;

        public TransitionOrder(int transitionGroupId, int? iteration, int? timestep, double? order)
        {
            this.m_TransitionGroupId = transitionGroupId;
            this.m_Iteration = iteration;
            this.m_Timestep = timestep;

            if (order.HasValue)
            {
                this.m_Order = order.Value;
            }
        }

        public int TransitionGroupId
        {
            get
            {
                return this.m_TransitionGroupId;
            }
        }

        public int? Iteration
        {
            get
            {
                return this.m_Iteration;
            }
        }

        public int? Timestep
        {
            get
            {
                return this.m_Timestep;
            }
        }

        public double Order
        {
            get
            {
                return this.m_Order;
            }
        }
    }
}
